### PR TITLE
DAOS-6796 control: dont forward rank down event with a nil-rank

### PR DESCRIPTION
**Pull request by tanabarr**
If engine crashes before it has joined the system then it won't yet have
been assigned a rank. In that case there is no point forwarding the
event to the MS leader because the membership cannot be updated.
If no valid rank exists when raising RankDown event, disable forwarding
for the event.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by kjacque — APPROVED** [review]
LGTM

File: src/control/events/ras.go [hunk 1]
```diff
@@ -165,6 +179,8 @@ func New(evt *RASEvent) *RASEvent {
 	if evt.Severity == RASSeverityUnknown {
 		evt.Severity = RASSeverityInfo
 	}
+	evt.forwarded = atm.NewBool(false)
+	evt.forwardable = atm.NewBool(true)
```
Comment (mjmac) [inline src/control/events/ras.go:183]: Why not `evt.forwarded.SetFalse()` and `evt.forwardable.SetTrue()`?

Every one of these `NewBool()` calls is an unsynchronized write to a variable. The zero value of an atm.Bool is already usable and set to False.

Looking at the atm API, I do see that there's a gap in that we have a `Load()` but not a `Store()`. I'll fix that.

Comment (tanabarr) [reply src/control/events/ras.go:183]: I wasn't clear on the uninitialized state of the variable, I will make those changes. I did notice the lack of Store() , thanks.

**Review by mjmac — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by tanabarr** [comment]
1 intermittent failure in CI due to https://jira.hpdd.intel.com/browse/DAOS-6319

**Pull request merged**